Hakawai/Mentions/HKWMentionsCreationStateMachine.m

```diff
@@ -756,6 +756,37 @@ - (void)dataReturnedWithResults:(NSArray *)results
     }
 }
 
+/*!
+ Handle updated data received from the data source.
+ */
+- (void)dataReturnedWithEmptyResults:(BOOL)isEmptyResults
+         keystringEndsWithWhiteSpace:(BOOL)keystringEndsWithWhiteSpace {
+    if (self.state == HKWMentionsCreationStateQuiescent && self.searchType != HKWMentionsSearchTypeInitial) {
```

Comment:
is this all new logic or is it copied from somewhere? 

why are we creating a whole new function?

is the comment above this accurate?